app/services/vacancy_search_builder.rb

```diff
@@ -79,15 +74,23 @@ def keyword_query
     end
   end
 
-  def location_geo_distance
+  def schools_in_area
+    SchoolsInArea.new(lat: geocoded_location[0],
+                      lng: geocoded_location[1],
+                      radius: radius).results.pluck(:id)
+  end
+
+  def schools_query
+    return if geocoded_location.nil?
+
     {
-      geo_distance: {
-        distance: "#{radius}mi",
-        coordinates: {
-          lat: geocoded_location.first,
-          lon: geocoded_location.last
-        }
-      }
+      bool: {
+        filter: {
+          terms: {
+            school_id: schools_in_area,
+          },
+        },
+      },
     }
```

Comment:
This looks a little weird to me. This results in something that looks like:
```
{
  bool: {
    must: [
      bool: {
        filter: {
          terms: {
            school_id: schools_in_area,
          }
        }
      },
      ...
    ]
  }
}
```
Whereas, `filter` used to be set on the first `bool`. Was that intentional?

Reply:
I think (and correct me if I'm wrong) that we were searching by all the search criteria, and then filtering by distance after doing the initial search. Now we're doing the reverse and the school ids are now things to be searched on, rather than filtered by. My ElasticSearch knowledge is not the best, so I could be wrong...

Reply:
The difference between `must` and `filter` is only whether the score is important. `must` uses the score, while `filter` discards it. If they're on the same level, I believe they're run concurrently, and it's a bit of a simpler read than having nested `bool` queries, IMO. I _think_ the functionality is the same, though, but I'm not sure.

https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-bool-query.html